src/OpenTelemetry.AutoInstrumentation.Native/continuous_profiler.h

```diff
@@ -35,6 +35,64 @@ extern "C"
     // ReSharper restore CppInconsistentNaming
 }
 
+namespace continuous_profiler
+{
+struct FunctionIdentifier
+{
+    mdToken  function_token;
+    ModuleID module_id;
+    bool     is_valid;
+
+    bool operator==(const FunctionIdentifier& p) const
+    {
+        return function_token == p.function_token && module_id == p.module_id && is_valid == p.is_valid;
+    }
+};
+
+struct FunctionIdentifierResolveArgs
+{
+    FunctionID  function_id;
+    COR_PRF_FRAME_INFO frame_info;
+
+    bool operator==(const FunctionIdentifierResolveArgs& p) const
+    {
+        return function_id == p.function_id && frame_info == p.frame_info;
+    }
+};
+}
+
```

Comment:
The hash needs to be revisited 
1. Simple xor will not give good distribution. 
2. Also, XOR being commutative following two will produce same hash value FunctionIdentifier{token=0x06000123, moduleId=0x00000001, is_valid=true}
FunctionIdentifier{token=0x00000001, moduleId=0x06000123, is_valid=true}
2. Ignoring is_valid will also produce collisions 
3.

Reply:
Something like this 

template <typename... Args>
std::size_t hash_combine(const Args&... args) {
    std::size_t seed = 0;
    
    (..., (seed ^= std::hash<Args>{}(args)+0x9e3779b9 + (seed << 6) + (seed >> 2)));
    return seed;
}

Uses left shift, right shift to prevent commutative collisions. That magic number is, well math magic to make hash values well spread.